azure-active-directory/microsoft-graph-api-signin-activity-reports.md

```diff
@@ -158,94 +167,39 @@ if ($null -ne $authResult.AccessToken) {
-            correlationId,
-            conditionalAccessStatus,
-            isRisky,
-            riskLevel,
- 
-            @{Name = 'status.errorCode'; Expression = {$_.status.errorCode}},
-            @{Name = 'status.failureReason'; Expression = {$_.status.failureReason}},
-            @{Name = 'status.additionalDetails'; Expression = {$_.status.additionalDetails}},
- 
-            @{Name = 'deviceDetail.deviceId'; Expression = {$_.deviceDetail.deviceId}},
-            @{Name = 'deviceDetail.displayName'; Expression = {$_.deviceDetail.displayName}},
-            @{Name = 'deviceDetail.operatingSystem'; Expression = {$_.deviceDetail.operatingSystem}},
-            @{Name = 'deviceDetail.browser'; Expression = {$_.deviceDetail.browser}},
- 
-            @{Name = 'location.city'; Expression = {$_.location.city}},
-            @{Name = 'location.state'; Expression = {$_.location.state}},
-            @{Name = 'location.countryOrRegion'; Expression = {$_.location.countryOrRegion}},
-            @{Name = 'location.geoCoordinates.altitude'; Expression = {$_.location.geoCoordinates.altitude}},
-            @{Name = 'location.geoCoordinates.latitude'; Expression = {$_.location.geoCoordinates.latitude}},
-            @{Name = 'location.geoCoordinates.longitude'; Expression = {$_.location.geoCoordinates.longitude}}
- 
-            for ($k = 0; $k -lt $canumbers; $k++) {
-                $temp = $thisEvent.conditionalAccessPolicies[$k].id
-                $eachEvent = $eachEvent | Add-Member @{"conditionalAccessPolicies.id$k" = $temp} -PassThru
- 
-                $temp = $thisEvent.conditionalAccessPolicies[$k].displayName
-                $eachEvent = $eachEvent | Add-Member @{"conditionalAccessPolicies.displayName$k" = $temp} -PassThru
- 
-                $temp = $thisEvent.conditionalAccessPolicies[$k].enforcedGrantControls
-                $eachEvent = $eachEvent | Add-Member @{"conditionalAccessPolicies.enforcedGrantControls$k" = $temp} -PassThru
- 
-                $temp = $thisEvent.conditionalAccessPolicies[$k].enforcedSessionControls
-                $eachEvent = $eachEvent | Add-Member @{"conditionalAccessPolicies.enforcedSessionControls$k" = $temp} -PassThru
- 
-                $temp = $thisEvent.conditionalAccessPolicies[$k].result
-                $eachEvent = $eachEvent | Add-Member @{"conditionalAccessPolicies.result$k" = $temp} -PassThru
-            }
-            $data += $eachEvent
+        for ($j = 0; $j -lt $myReportValue.Count; $j++) {
+            $data += $myReportValue[$j]
         }
         
         #
-        #Get url from next link
+        # Get url from next link
         #
         $url = ($myReport.Content | ConvertFrom-Json).'@odata.nextLink'
-    }while ($url -ne $null)
+    } while ($null -ne $url)
 }
 else {
     Write-Host "ERROR: No Access Token"
 }
- 
-$data | Sort -Property createdDateTime  | Export-Csv $outfile -encoding "utf8" -NoTypeInformation
+
+$data | Out-File -FilePath $outfile
```

Comment:
If you want to export csv, it might be unchanged.

```diff
- $data | Out-File -FilePath $outfile
+ $data | Sort -Property createdDateTime  | Export-Csv $outfile -encoding "utf8" -NoTypeInformation
```

Reply:
to export json file

```powershell
$data | ConvertTo-Json | Out-File -FilePath $outfile
```

Reply:
This one is also accepted.